src/App.tsx

```diff
@@ -15,47 +16,125 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort((item1, item2) => item1.length - item2.length);
+  }
+
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort();
+  }
```

Comment:
You can use 1 `sort ` method instead and change condition of sort inside callback 

Let's change it to `switch` case